### PR TITLE
fix: update broken links and throw errors when a broken link is found

### PR DESCRIPTION
**Pull request by tomfrenken**
## What Has Changed?

With this PR, docusaurus will throw an error when a broken link is introduced.
Additionally, some broken links were updated.

### CODE REVIEW

File: docusaurus.config.js [hunk 1]
```diff
@@ -20,7 +20,7 @@ module.exports = {
     'The one-stop shop for developing and extending SAP applications in the cloud.',
   url: 'https://sap.github.io',
   baseUrl: '/cloud-sdk/',
-  onBrokenLinks: 'warn',
+  onBrokenLinks: 'throw',
   onBrokenMarkdownLinks: 'warn',
```
Comment (deekshas8) [inline docusaurus.config.js:24]: [q] Can we also set `onBrokenMarkdownLinks` to throw so broken markdown links are also covered?

Comment (tomfrenken) [reply docusaurus.config.js:24]: I guess that would also make sense 🤔

Comment (tomfrenken) [reply docusaurus.config.js]: I've added it ✅

**Review by deekshas8 — APPROVED** [review]
LGTM.

**Pull request merged**